vta/hardware/chisel/src/main/scala/core/TensorStore.scala

```diff
@@ -62,20 +62,40 @@ class TensorStore(tensorType: String = "none", debug: Boolean = false)(
   val tag = Reg(UInt(8.W))
   val set = Reg(UInt(8.W))
 
+  // Dynamically adjust the size of DMA transfers to avoid crossing page boundaries.
+  final val ADAPTIVE_DMA_XFER_ENABLE = true
```

Comment:
Just curious -- is there a way this is false?. If this is fixing the wraparound AXI bug, then we should make it default and update all the code that depend on it. For example:

```
val xfer_init_bytes = if (ADAPTIVE_DMA_XFER_ENABLE) xmax_bytes - xfer_init_addr % xmax_bytes else xmax_bytes
```

will be replaced by

```
val xfer_init_bytes =  xmax_bytes - xfer_init_addr % xmax_bytes
```


Same in `TensorUtil.scala`

Reply:
There is small cost that the fix adds in terms of timing, which could be mitigated with a refactoring of the FSM.
I was not sure whether the wraparound problem is a systematic limitation or not so I decided to add it in a parametric way just in case another platform does no exhibits the issue.
I left the static constant there but one could drive it from the Configs.scala file if needed. 
The idea is to turn it off when trying another platform and if successful set the parameter to false in a config file.
Does it make sense?

Reply:
I see, AFAIK the problem is systematic across all platforms, so we might want to fix it right away. Also, I believe `de10-nano` is `AXI3`, therefore other platforms using `AXI4` should also work with this because AXI's _"backwards compatibility."_

Later, we can optimize for performance if we would like to.

Reply:
Well, this is a limitation, not a feature though, unless you are talking about bug backward compatibility. 
Indeed, you may very well have this limitation removed in future versions of AXI or maybe be using other interconnects that do not have it, and you may choose to take advantage of better performance.
That said, I don't mind taking it out, it will save two lines of scala code ;-)

Reply:
Yeah, I think we can remove it here and in `TensorUtil.scala`.

Reply:
Ok, will do shortly!